### PR TITLE
 SLI-839 Run actions update in a background thread 

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

**Comment by damien-urruty-sonarsource** [comment]
@henryju you can maybe review commit by commit. My idea was to make sure all of our actions have a common ancestor, but I had to keep 3 different ones: one for classic actions, one for ToggleActions and one for Checkbox Actions. But then the real fix was much easier, because I only had to modify those 3 places

File: src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java [hunk 1]
```diff
@@ -21,40 +21,43 @@
...
+import com.intellij.openapi.project.DumbAware;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import java.util.stream.Stream;
 import javax.swing.Icon;
 import org.jetbrains.annotations.Nullable;
 import org.sonarlint.intellij.analysis.AnalysisStatus;
 
-public abstract class AbstractSonarAction extends AnAction {
+public abstract class AbstractSonarAction extends AnAction implements DumbAware {
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java]: Have you anticipated the impacts of implementing `DumbAware`? Are we sure all or actions can work properly in Dumb mode?

**Review by henryju — APPROVED** [review]
Minor doubt about the impact of DumAware on all actions, but overall this is a very good move, well done.

File: src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/actions/AbstractSonarAction.java]: I tried the actions that are available with Shift-Shift when the project is in dumb mode. For all other actions they are available through the tool window and during indexing it is not shown. So I think it should be safe

**Pull request merged**